File: 1.java
import java.util.Scanner;

class TermDocumentMatrix {
    private int[][] matrix;
    private int numTerm;
    private int numDoc;

    public TermDocumentMatrix(int numTerm, int numDoc) {
        this.numTerm = numTerm;
        this.numDoc = numDoc;
        matrix = new int[numTerm][numDoc];
    }

    public void updateTermFrequency(int term, int doc, int val) {
        if (term < numTerm && doc < numDoc) {
            matrix[term][doc] = val;
        } else {
            System.out.println("Please enter valid index of document or term");
        }
    }

    public int getTermFrequency(int term, int doc) {
        if (term < numTerm && doc < numDoc) {
            return matrix[term][doc];
        } else {
            System.out.println("Please enter valid index of document or term");
            return -1;
        }
    }
}

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter the number of terms: ");
        int numTerm = scanner.nextInt();
        System.out.print("Enter the number of documents: ");
        int numDoc = scanner.nextInt();

        TermDocumentMatrix matrix = new TermDocumentMatrix(numTerm, numDoc);

        int choice;
        do {
            System.out.println("\nMenu:");
            System.out.println("1. Update Term Frequency");
            System.out.println("2. Get Term Frequency");
            System.out.println("0. Exit");
            System.out.print("Enter your choice: ");
            choice = scanner.nextInt();

            switch (choice) {
                case 1:
                    System.out.print("Enter term index: ");
                    int term = scanner.nextInt();
                    System.out.print("Enter document index: ");
                    int doc = scanner.nextInt();
                    System.out.print("Enter value: ");
                    int val = scanner.nextInt();
                    matrix.updateTermFrequency(term, doc, val);
                    break;
                case 2:
                    System.out.print("Enter term index: ");
                    term = scanner.nextInt();
                    System.out.print("Enter document index: ");
                    doc = scanner.nextInt();
                    System.out.println("Term frequency: " + matrix.getTermFrequency(term, doc));
                    break;
                case 0:
                    System.out.println("Exiting...");
                    break;
                default:
                    System.out.println("Invalid choice");
            }
        } while (choice != 0);

        scanner.close();
    }
}

File: 1.py
class TermDocumentMatrix:
    def __init__(self, num_term, num_doc):
        self.num_term = num_term
        self.num_doc = num_doc
        self.matrix = [[0 for _ in range(num_doc)] for _ in range(num_term)]

    def update_term_frequency(self, term, doc, val):
        if 0 <= term < self.num_term and 0 <= doc < self.num_doc:
            self.matrix[term][doc] = val
        else:
            print("Please enter valid index of document or term")

    def get_term_frequency(self, term, doc):
        if 0 <= term < self.num_term and 0 <= doc < self.num_doc:
            return self.matrix[term][doc]
        else:
            print("Please enter valid index of document or term")

def main():
    num_term = int(input("Enter the number of terms: "))
    num_doc = int(input("Enter the number of documents: "))
    matrix = TermDocumentMatrix(num_term, num_doc)

    while True:
        print("\nMenu:")
        print("1. Update Term Frequency")
        print("2. Get Term Frequency")
        print("0. Exit")
        choice = int(input("Enter your choice: "))

        if choice == 1:
            term = int(input("Enter term index: "))
            doc = int(input("Enter document index: "))
            val = int(input("Enter value: "))
            matrix.update_term_frequency(term, doc, val)
        elif choice == 2:
            term = int(input("Enter term index: "))
            doc = int(input("Enter document index: "))
            print(f"Term frequency: {matrix.get_term_frequency(term, doc)}")
        elif choice == 0:
            print("Exiting...")
            break
        else:
            print("Invalid choice")

if __name__ == "__main__":
    main()
